java/code/src/com/redhat/rhn/manager/system/VirtualInstanceManager.java

```diff
@@ -96,7 +96,13 @@ public static void updateGuestsVirtualInstances(Server server, List<VmInfo> plan
             int vCpus = info.getGuestProperties().getVcpus();
             long memory = info.getGuestProperties().getMemorySize();
 
-            uuid = fixUuidIfSwappedUuidExists(uuid);
+            try {
+                uuid = fixUuidIfSwappedUuidExists(uuid);
+            }
+            catch (IllegalArgumentException e) {
+                // Systems running in Public Clouds are able to reporting a real "uuid"
```

Comment:
I do not understand this comment?

Why does it throw IllegalArgumentException? How would uuid look like?

Reply:
Actually, the comment is not good. I'll fix it.

The `fixUuidIfSwappedUuidExists` method is expecting a UUID with format "xxxx-xxxx-xxxx-xxxxxxxx`. In case the "uuid" doesn't have that format, an `IllegalArgumentException` is raised.

Reply:
Wouldn't it be better to catch this inside of the method and just return the same string if the format does not match?